util.go

```diff
@@ -17,26 +30,42 @@ func generateSigner() (ssh.Signer, error) {
 }
 
 func parsePtyRequest(s []byte) (pty Pty, ok bool) {
-	term, s, ok := parseString(s)
-	if !ok {
-		return
-	}
-	width32, s, ok := parseUint32(s)
-	if !ok {
-		return
+	reqMsg := &ptyRequestMsg{}
+	err := ssh.Unmarshal(s, reqMsg)
+	if err != nil {
+		return Pty{}, false
 	}
-	height32, _, ok := parseUint32(s)
-	if !ok {
-		return
+
+	modes := []byte(reqMsg.Modelist)
+
+	mode := struct {
+		Key uint8
+		Val uint32
+	}{}
+
+	TerminalModes := make(ssh.TerminalModes, 0)
+	for {
+		if len(modes) < 1 || modes[0] == ttyOPEND || len(modes) < 5 {
+			break
+		}
+		b := modes[:5]
+		err = ssh.Unmarshal(b, &mode)
+		if err != nil {
+			return Pty{}, false
+		}
+		TerminalModes[mode.Key] = mode.Val
+		modes = modes[6:]
 	}
+
 	pty = Pty{
-		Term: term,
+		Term: reqMsg.Term,
 		Window: Window{
-			Width:  int(width32),
-			Height: int(height32),
+			Width:  int(reqMsg.Columns),
+			Height: int(reqMsg.Rows),
 		},
+		TerminalModes: TerminalModes,
 	}
-	return
+	return pty, true
```

Comment:
These are already implicitly returned because of the named params. Please either switch back to the implicit return or remove the variable names from the function definition.

Reply:
👌